core/src/main/java/hudson/model/UpdateSite.java

```diff
@@ -646,6 +661,144 @@ public Api getApi() {
 
     }
 
+    /**
+     * A version range for {@code Warning}s indicates which versions of a given plugin are affected
+     * by it.
+     *
+     * Name, firstVersion and lastVersion fields are only used for administrator notices.
+     *
+     * The pattern is used to determine whether a given warning applies to the current installation.
+     */
+    public static final class WarningVersionRange {
+        /**
+         * Human-readable English name for this version range, e.g. 'regular', 'LTS', '2.6 line'.
+         *
+         * Can be null.
+         */
+        public final String name;
+
+        /**
+         * First version in this version range to be subject to the warning. Can be null.
+         */
+        public final String firstVersion;
+
+        /**
+         * Last version in this version range to be subject to the warning. Can be null.
+         */
+        public final String lastVersion;
+
+        /**
+         * Regular expression pattern for this version range that matches all included version numbers.
+         */
+        @Nonnull
+        public final Pattern pattern;
+
+        public WarningVersionRange(JSONObject o) {
+            this.name = Util.fixEmpty(o.optString("name"));
+            this.firstVersion = Util.fixEmpty(o.optString("firstVersion"));
+            this.lastVersion = Util.fixEmpty(o.optString("lastVersion"));
+            this.pattern = Pattern.compile(o.getString("pattern"));
+        }
+    }
+
+    /**
+     * Represents a warning about a certain component, mostly related to known security issues.
+     *
+     * Two objects are considered equal if they have the same ID.
+     */
+    public static final class Warning {
+
+        public static final String TYPE_PLUGIN = "plugin";
+        public static final String TYPE_CORE = "core";
+        public static final String NAME_CORE = "core";
+
+        /**
+         * The type classifier for this warning. Two are currently supported:
+         * - core, for Jenkins itself
+         * - plugin, for Jenkins plugins
+         * Never null.
+         */
+        // TODO Do we actually need this?
+        @Exported
+        public final String type;
+
+        /**
+         * The globally unique ID of this warning.
+         *
+         * This is typically the CVE identifier or SECURITY issue (Jenkins project);
+         * possibly with a unique suffix (e.g. artifactId) if either applies to multiple components.
```

Comment:
Right, the example does not follow this format.

The idea as documented is that there could be `SECURITY-1234.core` and `SECURITY-1234.mailer`. The companion PR https://github.com/jenkinsci/backend-update-center2/pull/96 contains real data, but this case doesn't apply.